### ISSUE TITLE
Error during apply new tags for vpc endpoint services

### ISSUE THREAD

**Issue by KVInventoR** [terraform-providers/terraform-provider-aws#11675]
Hi All,

I use latest version of terraform aws provider for terraform 11

`
Terraform v0.11.14
+ provider.archive v1.3.0
+ provider.aws v2.44.0
+ provider.external v1.2.0
+ provider.local v1.4.0
+ provider.null v2.1.2
+ provider.template v2.1.2
`

```
module.wh.aws_vpc_endpoint.ec2: Still modifying... (ID: vpce-0e35874509952b, 45m11s elapsed)
module.wh.aws_vpc_endpoint.sns: Still modifying... (ID: vpce-0ddd82150dd7d2, 45m0s elapsed)
module.wh.aws_vpc_endpoint.sqs: Still modifying... (ID: vpce-0caa6a84c87304, 45m21s elapsed)
module.wh.aws_vpc_endpoint.ec2: Still modifying... (ID: vpce-0e35874509952b, 45m21s elapsed)
module.wh.aws_vpc_endpoint.sns: Still modifying... (ID: vpce-0ddd82154dd7d2, 45m10s elapsed)
module.wh.aws_vpc_endpoint.sqs: Still modifying... (ID: vpce-0caa6a84c87304, 45m31s elapsed)
module.wh.aws_vpc_endpoint.ec2: Still modifying... (ID: vpce-0e35874509952b, 45m31s elapsed)
module.wh.aws_vpc_endpoint.sns: Still modifying... (ID: vpce-0ddd82154dd7d2, 45m20s elapsed)
module.wh.aws_vpc_endpoint.sqs: Still modifying... (ID: vpce-0caa6a84c87304, 45m41s elapsed)
module.wh.aws_vpc_endpoint.ec2: Still modifying... (ID: vpce-0e35874509952b, 45m41s elapsed)
module.wh.aws_vpc_endpoint.sns: Still modifying... (ID: vpce-0ddd82154dd7d2, 45m30s elapsed)
```

in  the same time in cloudtrail logs I found:
```
1  2020-01-20T13:36:25Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com 
2  2020-01-20T13:37:46Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com 
3  2020-01-20T13:37:18Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
4  2020-01-20T13:38:06Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
5  2020-01-20T13:36:19Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
6  2020-01-20T13:37:26Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
7  2020-01-20T13:36:38Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
8  2020-01-20T13:36:31Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
9  2020-01-20T13:37:35Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
10 2020-01-20T13:36:44Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
11 2020-01-20T13:41:03Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
12 2020-01-20T13:46:59Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
13 2020-01-20T14:06:19Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
14 2020-01-20T14:01:55Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
15 2020-01-20T13:43:33Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
16 2020-01-20T13:55:40Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
17 2020-01-20T14:12:48Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
18 2020-01-20T14:09:53Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
19 2020-01-20T13:39:06Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
20 2020-01-20T13:51:59Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
21 2020-01-20T13:58:38Z ModifyVpcEndpoint ADFS-Techops Server.InternalError ec2.amazonaws.com
22 2020-01-20T13:36:52Z ModifyVpcEndpoint ADFS-Techops Client.RequestLimitExceeded ec2.amazonaws.com
```

Our code just only did next: 
Changed 1 of tag on VPC PrivateLink from 1 value to another.
The most intersting part, issue present only in a couple of vpc, 
for 8 vpc we applied changed without any issues
for 2 vpc we got an issue.
Also we tried to make this changes during all week, more that 10 times.

I found a similar issue:
https://github.com/terraform-providers/terraform-provider-aws/pull/3418
but looks like issue was closed

Does anybody has the same issue?

**Comment by ewbankkit** [comment]
@KVInventoR Thanks for raising this issue.
I see the problem here. `ModifyVpcEndpoint` is being called even if it's just the `tags` that have changed on an endpoint:

https://github.com/terraform-providers/terraform-provider-aws/blob/990fefeae076ff8d7576a8b031028a64ab4b42b2/aws/resource_aws_vpc_endpoint.go#L287-L324

**Comment by bflad** [comment]
The fix to skip calling `ModifyVpcEndpoint` on tags only updates has been merged and will release with version 2.68.0 of the Terraform AWS Provider, later this week. Thanks to @ewbankkit for the implementation. 👍
